### DISCUSSION TITLE
v4 - styled mode - custom theme - severity / buttons colors

### DISCUSSION THREAD

**Discussion by bobby-driggs** [primefaces/community#2095] [PrimeVue]
Looking at the existing Lara theme for v4 - beta4, it seems the severity colors are directly using the `primitive` color tokens.

https://github.com/primefaces/primevue/blob/e275149c3d2f74d6b8fcaff057f30023178a3102/components/lib/themes/lara/button/index.js#L65

`sky.500` for info, `green.500` for success, etc.

I was expecting there to be `semantic` tokens, similar to `surface.XXX` for severity, like `info.XXX` or `success.XXX`.

Our designer has an entirely different color pallet, and I brought it in as `primitive` tokens, but it seems like i need to map their color names to the existing `primitive` names, and use their severity color pallet as named `primitive` tokens. 

This seems awkard, is there a better way?

--

Edit: I can obv override the button components styling, to use a custom semantic token, but it also seems like i would need to override multiple components to get this right throughout.

**Comment by leo-ay** [comment]
Indeed!

I thought we could customize the `info` pallete color shade, for example, exactly as we can with the `primary` pallete.
In other words, setting the shade and it compiles across every component that have the `info` variation (message, button, tag, badge etc).

There is a ugly global workaround that is to change the primitive colors. If you want all `info` components to be red, we can change the `sky` palettes to have red colors lol. It pretty much defeats the concept of primitive layer to semantic layer.

I hope the Devs could look into it, because it would reinforce their concept for the theme architeture:
https://primevue.org/theming/styled/#architecture

Complementary to that, if the Design Tokens will be the only way to customize this, if the Documenation mentioned the shade being used by default - example for button info light:
```
background: '{xxx.500}',
hoverBackground: '{xxx.600}',
activeBackground: '{xxx.700}',
 ...
```
 it would helpful for us to keep the customization consistent with default compononents (in case we want to change only one or another severity variation)

**Reply by bobby-driggs** [reply]
My current workaround is that i made my own `semantic` tokens for things like `info`, `warning`, etc, and set them based on my `primitive` color tokens.

I searched the theme, and found the components (button, tag, badge, progress-spinner) that were using the base `sky` tokens, and swapped them to use my own `semantic` tokens.

Basically how i originally would have expected it all to work

**Comment by vieira-brz** [comment]
Alright, let's break down this color token business in a way that's easier to grasp, without losing the technical bit.

The problem is that the Lara theme for PrimeVue directly uses the system colors, like "sky.500" for "info" and "green.500" for "success". This is a bit of a pain because:

1. **Everything's tied together:**  If you want to use your own colors, you've got to mess with the whole theme, which is a drag!
2. **Not flexible:** Want to add a new severity type (like "warning")? You'll have to touch the theme's code again! 

**Better solutions to consider:**

1. **Create your own semantic tokens:**

   - **Define your tokens:**  Instead of "info", use "your-info-500", "your-info-300" and so on.
   - **Make a map:**  Create a file where you link your tokens to your design system's colors, like "your-info-500"  -> "yourDesignSystemInfo500".
   - **Use them in components:**  In your CSS classes, use your tokens. The map will work its magic to find the right color.

2. **Extend the Lara theme:**

   - **Start with Lara:** Grab the Lara theme's code and make a copy.
   - **Swap out the colors:** In the theme code, replace the button, input, etc. colors with your own.

3. **Quick fixes:**

   - **Direct CSS:** If you just need to change a component's color, you can use direct CSS. But this can get messy if you have to change lots of components!
   - **Custom classes:** Create CSS classes with your styles and apply them to components. 

**What should you choose?**

* **Semantic tokens:** Best for large systems that change often. It's more work upfront, but pays off in the long run.
* **Extend the theme:** Great if you want to quickly customize the Lara theme with your colors.
* **Quick fixes:** For one-off tweaks, but can become a Frankenstein monster if you're not careful.

**Keep in mind:**

* **Have a design system documented:** Documenting colors, typography, etc., will help you maintain consistency!
* **Start simple:** Try one method and see how it works. Then, refine it as your project grows.
